### PR TITLE
Add regression in spline trainer

### PR DESCRIPTION
**Pull request by deerzq**
Add regression for spline trainer
@hectorgon

### CODE REVIEW

File: training/src/main/scala/com/airbnb/aerosolve/training/SplineTrainer.scala [hunk 1]
```diff
@@ -605,6 +615,21 @@ object SplineTrainer {
     return loss
   }
 
+  def updateRegressor(model: SplineModel,
+                      fv: FeatureVector,
+                      label: Double,
+                      params : SplineTrainerParams) : Double = {
+    val flatFeatures = Util.flattenFeatureWithDropout(fv, params.dropout)
+    val prediction = model.scoreFlatFeatures(flatFeatures) / (1.0 - params.dropout)
+    val loss = math.abs(prediction - label) // absolute difference
```
Comment (hectorgon) [inline training/src/main/scala/com/airbnb/aerosolve/training/SplineTrainer.scala]: Isn't it max(0, abs(p-l) - epsilon) ? But this is just reporting the loss so no big deal whichever you think seems better.

**Comment by hectorgon** [comment]
LGTM
